synapse/storage/database.py

```diff
@@ -1483,6 +1486,9 @@ def simple_upsert_many_txn(
             value_values: A list of each row's value column values.
                 Ignored if value_names is empty.
         """
+        if not key_values:
```

Comment:
```suggestion
        if not key_values and not value_values:
```

I think `key_names` and `key_values` have to be empty to make sense, but still

Reply:
I'm not sure what you mean; I guess this could be used as:

```python
db_pool.simple_upsert_many_txn(
    txn,
    table="foo",
    key_names: (),
    key_values: [(), ()],
    value_names: ("a",),
    value_values: [(1,), (2,)],
)
```

tl;dr I think `key_values` and `value_values` need to be the same length?

Reply:
Ah, OK fair.

Reply:
It looks like you *can* provide only `key_values` and not `value_values` and we'll automatically fill those in, so I *think* this is the correct change to make.